Index: src/static_Instance/StaticInstanceBlock.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/static_Instance/StaticInstanceBlock.java b/src/static_Instance/StaticInstanceBlock.java
new file mode 100644
--- /dev/null	(date 1770629117526)
+++ b/src/static_Instance/StaticInstanceBlock.java	(date 1770629117526)
@@ -0,0 +1,31 @@
+package static_Instance;
+
+public class StaticInstanceBlock
+{
+    {
+        System.out.println(" before main instance block Initialized");
+    }
+    static
+    {
+        System.out.println("Static 1 block Initialized");
+    }
+    static
+    {
+        System.out.println("Static 2 block Initialized");
+    }
+    public StaticInstanceBlock()
+    {
+        System.out.println("Constructor Initialized");
+    }
+
+    public static void main(String[] args) {
+        new StaticInstanceBlock();
+    }
+    static
+    {
+        System.out.println(" After main Static block Initialized");
+    }
+    {
+        System.out.println("Instance block after main Initialized");
+    }
+}
Index: src/static_Instance/InstanceBlock.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/static_Instance/InstanceBlock.java b/src/static_Instance/InstanceBlock.java
new file mode 100644
--- /dev/null	(date 1770629117502)
+++ b/src/static_Instance/InstanceBlock.java	(date 1770629117502)
@@ -0,0 +1,22 @@
+package static_Instance;
+
+public class InstanceBlock
+{
+    {  //does not need to call it called automatically whenever constructor call
+        System.out.println("instance block Initialized before main first instance block");
+    }
+    {
+        System.out.println("instance block Initialized after main second instance block");
+    }//every instance block called before constructor
+    InstanceBlock()
+    {
+        System.out.println(" Constructor Initialized");
+    }
+    public static void main(String[] args) {
+        InstanceBlock ib =new InstanceBlock();
+        new InstanceBlock();//how many constructor call that many times instance block called
+    }
+    { //whenever we Initialized constructor that time instance block called once for
+        System.out.println("instance block Initialized after main");
+    }
+}
Index: src/static_Instance/StaticKeyword.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/static_Instance/StaticKeyword.java b/src/static_Instance/StaticKeyword.java
new file mode 100644
--- /dev/null	(date 1770629117536)
+++ b/src/static_Instance/StaticKeyword.java	(date 1770629117536)
@@ -0,0 +1,25 @@
+package static_Instance;
+
+public class StaticKeyword
+{
+    static//call first
+    {
+     System.out.println("Static 1 block Initialized");
+    }
+    static//we can write any number of static block
+    {
+        System.out.println("Static2 block Initialized");
+    }
+    static void meth1()//then method call when ever main call meth1
+    {
+       System.out.println("Static called meth1");
+    }
+    public static void main(String[] args)
+    {
+        meth1();//static method call
+    }
+    static //after first static block
+    {
+        System.out.println("Static 3 block ");
+    }
+}
